app/Models/Node.php

```diff
@@ -268,6 +270,11 @@ public function databaseHosts(): BelongsToMany
         return $this->belongsToMany(DatabaseHost::class);
     }
 
+    public function roles(): HasManyThrough
+    {
+        return $this->hasManyThrough(Role::class, NodeRole::class);
```

Comment:
```suggestion
        return $this->hasManyThrough(Role::class, NodeRole::class, 'node_id', 'id', 'id', 'role_id');
```
Default Pivot don't have `id` so we have to specify them.

I'd make `accessibleNodes` a QueryBuilder so we can use it in the controller like for `accessibleServers`
https://github.com/rmartinoscar/panel/tree/boy132/node-ownership